tutorials/Diagnostics/Debug/StateCheck.jl

```diff
@@ -0,0 +1,176 @@
+# # State debug statistics
+#
+# This page shows how to use the ```StateCheck``` functions to get basic
+# statistics for nodal values of fields held in ClimateMachine ```MPIStateArray```
+# data structures. The ```StateCheck``` functions can be used to
+#
+# 1. Generate statistics on ```MPIStateArray``` holding the state #    of a ClimateMachine experiment.
+# 
+#    and to
+#
+# 2. Compare against saved reference statistics from ClimateMachine ```MPIStateArray```
+#    variables. This can enable simple automated regression test checks for
+#    detecting unexpected changes introduced into numerical experiments
+#    by code updates.
+#
+# These two cases are shown below:
+
+
+# ## 1. Generating statistics for a set of MPIStateArrays
+#
+# Here we create a callback that can generate statistics for an arbitrary
+# set of the MPIStateArray type variables of the sort that hold persistent state for
+# ClimateMachine models. We then invoke the call back to show the statistics.
+#
+# In regular use the ```MPIStateArray``` variables will come from model configurations.
+# Here we create a dummy set of ```MPIStateArray``` variables for use in stand alone
+# examples.
+
+# ### Create a dummy set of MPIStateArrays
+#
+# First we set up two ```MPIStateArray``` variables. This need a few packages to be in placeT,
+# and utilizes some utility functions to create the array and add named
+# persistent state variables.
+# This is usually handled automatically as part of model deefinition in regular
+# ClimateMachine activity.
+
+# Set up a basic environment
+using MPI
+using StaticArrays
+using Random
+using ClimateMachine
+using ClimateMachine.VariableTemplates
+using ClimateMachine.MPIStateArrays
+using ClimateMachine.StateCheck
+if !MPI.Initialized()
+    MPI.Init()
+end
+FT = Float64
+
+# Define some dummy vector and tensor abstract variables with associated types
+# and dimensions
+F1 = @vars begin
+    ν∇u::SMatrix{3, 2, FT, 6}
+    κ∇θ::SVector{3, FT}
+end
+F2 = @vars begin
+    u::SVector{2, FT}
+    θ::SVector{1, FT}
+end
+nothing # hide
+
+# Create ```MPIStateArray``` variables with arrays to hold elements of the 
+# vectors and tensors
+Q1 = MPIStateArray{Float32, F1}(
+    MPI.COMM_WORLD,
+    ClimateMachine.array_type(),
+    4,
+    9,
+    8,
+)
+Q2 = MPIStateArray{Float64, F2}(
+    MPI.COMM_WORLD,
+    ClimateMachine.array_type(),
+    4,
+    6,
+    8,
+)
+nothing # hide
+
+# ### Create a call-back
+#
+# Now we can create a ```StateCheck``` call-back, _cb_, tied to the ```MPIStateArray```
+# variables _Q1_ and _Q2_. Each ```MPIStateArray``` in the array
+# of ```MPIStateArray``` variables tracked is paired with a label
+# to identify it. The call-back is also given a frequency (in time step numbers) and
+# precision for printing summary tables.
+cb = ClimateMachine.StateCheck.sccreate(
+    [(Q1, "My gradients"), (Q2, "My fields")],
+    1;
+    prec = 15,
+)
```

Comment:
This is outputting the following box in the docs:
```
# SC Start: creating state check callback
# SC Creating state check callback labeled "My gradients" for symbols
# SC ν∇u
# SC κ∇θ
# SC Creating state check callback labeled "My fields" for symbols
# SC u
# SC θ
# SC Finish: creating state check callback
```
I don't know if that is intended or not.

Reply:
It was intended. Not sure if it helps, but I'm inclined to leave it for now to see.